tests/main/xdg_runtime_dir/task.yaml

```diff
@@ -0,0 +1,40 @@
+summary: Ensure that XDG_RUNTIME_DIR is created
+
+details: |
+    Ensure that the XDG_RUNTIME_DIR is created when a snap is launched.
+# effectively we can only build the snap using LXD + snapcraft on ubuntu systems
+# due to various bugs with other distros and snapcraft and/or LXD
+systems:
+    # snapcraft is a classic snap, can't run on ubuntu-core
+    - -ubuntu-core*
+    # LXD doesn't work properly on amazon linux, fedora 31, opensuse, or debian
+    # 9 to build the snap, it complains about fuse.snapfuse getting EPERM when
+    # snaps are installed inside the LXD container that snapcraft creates
+    - -amazon-linux*
+    - -debian-*
+    - -fedora-*
+    - -opensuse-*
+    # additionally on fedora 32 and 33, and arch linux which are all cgroupsv2
+    # now, LXD does not work either, with the service failing to start up at all
+    # when we install LXD
+    - -arch-*
+    # snapcraft does not support running on centos 7 due to errors with the
+    # locale, see https://bugs.launchpad.net/snapcraft/+bug/1922140
+    # a separate error occurs on centos 8, see https://bugs.launchpad.net/snapcraft/+bug/1922981
+    - -centos-*
+    # ubuntu-18.04-32: i386 is not supported by lxd
+    # TODO: enable i386 by using lxd 3.0, currently snapcraft is failing to get the
+    #       lxd image when it is trying to build snapd
+    - -ubuntu-18.04-32
+
+
+execute: |
+    echo Testing XDG_RUNTIME_DIR
+    snap install snapcraft --classic
+    snap install lxd
+    lxd init --auto
+    snapcraft
+    snap install --dangerous xdg_test_0.1_*.snap
+    ls /run/user/
+    rm -f /run/user/0/snap.xdgtest
+    snap run xdgtest
```

Comment:
I think this test needs to be a bit more complete. There are no checks that actually verify whether or not the folder was created?

Reply:
The xdgtest snap returns an error if the folder doesn't exist...

Reply:
Added a comment explaining that.

Reply:
I would like to see an explicit check and echo with the error so it is more clean when the test fails.

Reply:
Ok.

Reply:
Done.